node/connector.ts

```diff
@@ -12,21 +12,55 @@ import {
   SettlementResponse,
   Settlements,
 } from '@vtex/payment-provider'
+import { VBase } from '@vtex/api'
 
 import { randomString } from './utils'
 import { executeAuthorization } from './flow'
 
+const authorizationsBucket = 'authorizations'
+const persistAuthorizationResponse = async (
+  vbase: VBase,
+  resp: AuthorizationResponse
+) => vbase.saveJSON(authorizationsBucket, resp.paymentId, resp)
+
+const getPersistedAuthorizationResponse = async (
+  vbase: VBase,
+  req: AuthorizationRequest
+) =>
+  vbase.getJSON<AuthorizationResponse | undefined>(
+    authorizationsBucket,
+    req.paymentId,
+    true
+  )
+
 export default class TestSuiteApprover extends PaymentProvider {
   // This class needs modifications to pass the test suit.
   // Refer to https://help.vtex.com/en/tutorial/payment-provider-protocol#4-testing
   // in order to learn about the protocol and make the according changes.
 
+  private async saveAndRetry(
+    req: AuthorizationRequest,
+    resp: AuthorizationResponse
+  ) {
+    await persistAuthorizationResponse(this.context.clients.vbase, resp)
+    this.callback(req, resp)
+  }
+
   public async authorize(
     authorization: AuthorizationRequest
   ): Promise<AuthorizationResponse> {
     if (this.isTestSuite) {
+      const persistedResponse = await getPersistedAuthorizationResponse(
+        this.context.clients.vbase,
+        authorization
+      )
+
+      if (persistedResponse !== undefined && persistedResponse !== null) {
```

Comment:
You can do something like `if (!!persistedResponse)` instead of this bigger expression or use `!isNil(persistedResponse)` in case you already using ramda is this project.